frontend/src/bundles/common/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+import { Navigate } from 'react-router-dom';
+
+import { Loader } from '~/bundles/common/components/components.js';
+import { AppRoute, DataStatus } from '~/bundles/common/enums/enums.js';
+
+import { useAppSelector } from '../../hooks/hooks.js';
+
+interface Properties {
+    children: React.ReactNode;
+}
+
+const ProtectedRoute: React.FC<Properties> = ({ children }) => {
+    const [loading, setLoading] = useState(true);
```

Comment:
```suggestion
    const [isLoading, setIsLoading] = useState(true);
```
B19
https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/src/javascript.md